lib/evmone/instructions_calls.cpp

```diff
@@ -435,8 +435,18 @@ Result create_eof_impl(
 
     if constexpr (Op == OP_TXCREATE)
     {
-        const auto error_subcont = validate_eof(state.rev, ContainerKind::initcode, initcontainer);
-        if (error_subcont != EOFValidationError::success)
+        const auto validity = state.get_initcode_validity(initcode_hash);
```

Comment:
This section is unnecessarily separated from the other `TXCREATE` processing section above. I think we should reorder them and place the endowment and depth checks before the `get_tx_initcode_by_hash()` section.

Then we will be able to combine the `get_tx_initcode_by_hash()` and `set_initcode_validity`, potentially with having single `unordered_map`.

Reply:
Problem is `pos` has to be updated before light failures
(and differently for EOFCREATE and TXCREATE)

I pushed one possible version, without unifying initcode and validity query yet.